frontend/src/utils/schedule-helpers.ts

```diff
@@ -310,3 +310,45 @@ export function getPositionOfYearInSchedule(
 export function isYearInPast(yearIndex: number, academicYear: number): boolean {
   return academicYear > yearIndex + 1;
 }
+
+/**
+ *  Determines if this course is in the given term
+ * @param courseToAdd the course that is being checked
+ * @param term the term being checked
+ * @returns whether or not this course is in the term
+ */
+function isCourseInTerm(courseToAdd: ScheduleCourse, term: DNDScheduleTerm) {
+  for (let course of term.classes) {
+    // courseToAdd's classId is an int, so we're casting in order to compare accurately
+    if (
+      String(courseToAdd.classId) === String(course.classId) &&
+      courseToAdd.subject === course.subject
+    ) {
+      return true;
+    }
+  }
+  return false;
+}
```

Comment:
Could we simplify this to avoid needing to explicitly return `true` or `false` based upon a condition?

```typescript
function isCourseInTerm(courseToAdd: ScheduleCourse, term: DNDScheduleTerm) {
    return term.classes.some((course) => String(courseToAdd.classId) === String(course.classId) && courseToAdd.subject === course.subject)
}
```

Reply:
yea that makes sense, this code was copied from the current add class component, probably want to clean this up for sure.